File: src/js/directable/directables/dotmatrix/Dots/Dot.cs
using UnityEngine;
using UnityEngine.UIElements;
using UnityTest.UserInterface.DotMatrix.Grid;

namespace UnityTest.UserInterface.DotMatrix.Dots
{
    public class Dot
    {
        private GameObject m_go;
        private SpriteRenderer m_spriteRenderer;
        private Sprite m_sprite;
        private Vector3 m_positionWorld = new Vector3(0, 0, 2);

        // _____________________________________________________________________

        public Dot(int dotIndex, GameObject container)
        {
            // Create Game Object
            m_go = new GameObject("DotMatrix_" + dotIndex);
            m_go.layer = LayerMask.NameToLayer("UserInterface");
            m_go.transform.position = new Vector3(0, 0, 0);
            m_go.transform.SetParent(container.transform);

            // Add Sprite Renderer
            m_spriteRenderer = m_go.AddComponent<SpriteRenderer>();

            // Create Sprite
            CreateSprite(new Vector2Int(GridData.GetGridWidth(), GridData.GetGridHeight()));

            // Load Sprite
            // LoadSprite("Sprites/DotMatrix/dot-white-64.png");

            // Start Cleared
            Clear();
        }

        // ______________________________________________________________ Update

        public void Update()
        {

        }

        // __________________________________________________________ Resolution

        public void SetResolution(Vector2Int resolution)
        {
            CreateSprite(resolution);
        }

        // ____________________________________________________________ Position

        public void SetPosition(Vector2 positionPixels)
        {
            // Store
            m_positionWorld.x = positionPixels.x;
            m_positionWorld.y = positionPixels.y;

            // Set
            m_go.transform.position = m_positionWorld;
        }

        // ________________________________________________________________ Fill

        public void Fill()
        {
            m_spriteRenderer.color = Color.white;
        }

        // _______________________________________________________________ Clear

        public void Clear()
        {
            m_spriteRenderer.color = Color.clear;
        }

        // ______________________________________________________________ Sprite

        private void CreateSprite(Vector2Int resolution)
        {
            // Create a new texture with the specified resolution
            Texture2D texture = new Texture2D(resolution.x, resolution.y);
            texture.filterMode = FilterMode.Point;

            // Fill with white pixels initially
            Color[] pixels = new Color[resolution.x * resolution.y];
            for (int i = 0; i < pixels.Length; i++)
                pixels[i] = Color.white;

            texture.SetPixels(pixels);
            texture.Apply();

            // Create and assign sprite
            m_sprite = Sprite.Create(texture,
                new Rect(0, 0, resolution.x, resolution.y),
                new Vector2(0.0f, 1.0f));

            m_spriteRenderer.sprite = m_sprite;
        }

        private void LoadSprite(string path)
        {
            // Load sprite from path
            m_sprite = Resources.Load<Sprite>(path);
            m_spriteRenderer.sprite = m_sprite;
        }
    }
}

File: src/js/directable/directables/dotmatrix/dot/Dot.js
export default class Dot {
	// private GameObject #go;
	// private SpriteRenderer #spriteRenderer;
	// private Sprite #sprite;
	// private Vector3 #positionWorld = new Vector3(0, 0, 2);

	// _________________________________________________________________________

	dot(dotIndex, container) {
		// // Create Game Object
		// #go = new GameObject("DotMatrix_" + dotIndex);
		// #go.layer = LayerMask.NameToLayer("UserInterface");
		// #go.transform.position = new Vector3(0, 0, 0);
		// #go.transform.SetParent(container.transform);
		// // Add Sprite Renderer
		// #spriteRenderer = #go.AddComponent<SpriteRenderer>();
		// // Create Sprite
		// CreateSprite(new Vector2Int(GridData.GetGridWidth(), GridData.GetGridHeight()));
		// // Load Sprite
		// // LoadSprite("Sprites/DotMatrix/dot-white-64.png");
		// // Start Cleared
		// Clear();
	}

	// __________________________________________________________________ Update

	update() {}

	// ______________________________________________________________ Resolution

	setResolution(resolution) {
		// CreateSprite(resolution);
	}

	// ________________________________________________________________ Position

	setPosition(positionPixels) {
		// // Store
		// #positionWorld.x = positionPixels.x;
		// #positionWorld.y = positionPixels.y;
		// // Set
		// #go.transform.position = #positionWorld;
	}

	// ____________________________________________________________________ Fill

	fill() {
		// #spriteRenderer.color = Color.white;
	}

	// ___________________________________________________________________ Clear

	clear() {
		// #spriteRenderer.color = Color.clear;
	}

	// __________________________________________________________________ Sprite

	#createSprite(resolution) {
		// // Create a new texture with the specified resolution
		// Texture2D texture = new Texture2D(resolution.x, resolution.y);
		// texture.filterMode = FilterMode.Point;
		// // Fill with white pixels initially
		// Color[] pixels = new Color[resolution.x * resolution.y];
		// for (int i = 0; i < pixels.Length; i++)
		//     pixels[i] = Color.white;
		// texture.SetPixels(pixels);
		// texture.Apply();
		// // Create and assign sprite
		// #sprite = Sprite.Create(texture,
		//     new Rect(0, 0, resolution.x, resolution.y),
		//     new Vector2(0.0, 1.0));
		// #spriteRenderer.sprite = #sprite;
	}

	#loadSprite(path) {
		// // Load sprite from path
		// #sprite = Resources.Load<Sprite>(path);
		// #spriteRenderer.sprite = #sprite;
	}
}
